core/trino-main/src/main/java/io/trino/util/DateTimeUtils.java

```diff
@@ -67,9 +69,69 @@ public static int parseDate(String value)
         // - the required format is 'YYYY-MM-DD'
         // - all components should be unsigned numbers
         // https://github.com/trinodb/trino/issues/10677
+        OptionalInt days = tryParseDateYyyyMmDd(value);
+        if (days.isPresent()) {
+            return days.getAsInt();
+        }
         return toIntExact(TimeUnit.MILLISECONDS.toDays(DATE_FORMATTER.parseMillis(value)));
     }
 
+    /**
+     * Check if the date is in format yyyy-MM-dd, if yes, return the epoch days; otherwise
+     * return OptionalInt.empty(), indicating the parse failed, and should fall back to generic
+     * parsing. Mixing the format checking and parsing in one pass for better performance.
+     * @param value date string to be parsed
+     * @return the number of days since 1970-01-01
+     */
+    public static OptionalInt tryParseDateYyyyMmDd(String value)
```

Comment:
`parseIfIso8861DateFormat`?

Reply:
renamed in d6f05bd8ec6e41ee32e70f8d6cfb128e447d74b0